nimbleclient/v1/restclient.py

```diff
@@ -237,6 +237,12 @@ def get(self, endpoint, filter=None, limit=None, from_id=None, **params):
                 if 'data' in response and len(response['data']) == 0:
                     break
 
+                # Add break condition when 'endRow' is equal to 'totalRows'.
+                # This is to handle 'controllers' list() operation in particular.
+                # The 'controllers' endpoint doesn't support 'startRow' or 'endRow' as valid query parameters in the URL.
```

Comment:
the comments should be corrected.. it is not specific to controller..
it can happen with other object set too.. This is a generic logic in a sense that when server returns response which has endRow == totalRow then we should break. as there is nothing more to fetch.

it can happen with simple calls like get() or can happen when objectset calls get(endRow=<for ex: 2>).. and assume that the object only had 2 entires.. 
both the calls will ultimately break here

Reply:
> the comments should be corrected.. it is not specific to controller..
> it can happen with other object set too.. This is a generic logic in a sense that when server returns response which has endRow == totalRow then we should break. as there is nothing more to fetch.
> 
> it can happen with simple calls like get() or can happen when objectset calls get(endRow=<for ex: 2>).. and assume that the object only had 2 entires..
> both the calls will ultimately break here

We intentionally wanted to avoid that explicit check so that we will make enough calls to the server until there are more records to fetch or server returns empty list.
Looks like its a bug on the array that it throwing error for only objectset 'controllers' and for other objects it is a valid filter (Whether they have 2 records or more records).

Reply:
Updated the comment to keep it generic.